lib/Remotery.c

```diff
@@ -4674,6 +4690,84 @@ RMT_API void _rmt_EndCPUSample(void)
+    { RMT_ERROR_WEBSOCKET_DISCONNECTED,       "RMT_ERROR_WEBSOCKET_DISCONNECTED - WebSocket server received a disconnect request and closed the socket" },
+    { RMT_ERROR_WEBSOCKET_BAD_FRAME_HEADER,   "RMT_ERROR_WEBSOCKET_BAD_FRAME_HEADER - Couldn't parse WebSocket frame header" },
+    { RMT_ERROR_WEBSOCKET_BAD_FRAME_HEADER_SIZE, "RMT_ERROR_WEBSOCKET_BAD_FRAME_HEADER_SIZE - Partially received wide frame header size" },
+    { RMT_ERROR_WEBSOCKET_BAD_FRAME_HEADER_MASK, "RMT_ERROR_WEBSOCKET_BAD_FRAME_HEADER_MASK - Partially received frame header data mask" },
+    { RMT_ERROR_WEBSOCKET_RECEIVE_TIMEOUT,    "RMT_ERROR_WEBSOCKET_RECEIVE_TIMEOUT - Timeout receiving frame header" },
+
+    { RMT_ERROR_REMOTERY_NOT_CREATED,       "RMT_ERROR_REMOTERY_NOT_CREATED - Remotery object has not been created" },
+    { RMT_ERROR_SEND_ON_INCOMPLETE_PROFILE, "RMT_ERROR_SEND_ON_INCOMPLETE_PROFILE - An attempt was made to send an incomplete profile tree to the client" },
+
+    // CUDA error messages
+    { RMT_ERROR_CUDA_DEINITIALIZED,         "RMT_ERROR_CUDA_DEINITIALIZED - CUDA driver is in the process of shutting down" },
+    { RMT_ERROR_CUDA_NOT_INITIALIZED,       "RMT_ERROR_CUDA_NOT_INITIALIZED - CUDA driver has not been initialized with cuInit() or that initialization has failed" },
+    { RMT_ERROR_CUDA_INVALID_CONTEXT,       "RMT_ERROR_CUDA_INVALID_CONTEXT - There is no context bound to the current thread" },
+    { RMT_ERROR_CUDA_INVALID_VALUE,         "RMT_ERROR_CUDA_INVALID_VALUE - One or more of the parameters passed to the API call is not within an acceptable range of values" },
+    { RMT_ERROR_CUDA_INVALID_HANDLE,        "RMT_ERROR_CUDA_INVALID_HANDLE - A resource handle passed to the API call was not valid" },
+    { RMT_ERROR_CUDA_OUT_OF_MEMORY,         "RMT_ERROR_CUDA_OUT_OF_MEMORY - The API call failed because it was unable to allocate enough memory to perform the requested operation" },
+    { RMT_ERROR_ERROR_NOT_READY,            "RMT_ERROR_ERROR_NOT_READY - A resource handle passed to the API call was not valid" },
+
+    // Direct3D 11 error messages
+    { RMT_ERROR_D3D11_FAILED_TO_CREATE_QUERY, "RMT_ERROR_D3D11_FAILED_TO_CREATE_QUERY - Failed to create query for sample" },
+
+    // OpenGL error messages
+    { RMT_ERROR_OPENGL_ERROR,               "RMT_ERROR_OPENGL_ERROR - Generic OpenGL error, no real need to expose more detail since app will probably have an OpenGL error callback registered" },
+
+    { RMT_ERROR_CUDA_UNKNOWN,               "RMT_ERROR_CUDA_UNKNOWN - Unknown CUDA error" }
+};
+
+RMT_API const char* _rmt_GetErrorString(rmtError errcode)
+{
+    int i;
+    for(i=0; i<sizeof(rmtErrorMappings)/sizeof(rmtErrorMappings[0]); ++i)
+    {
+        if(rmtErrorMappings[i].code == errcode)
+            return rmtErrorMappings[i].str;
+    }
+
+    return "Unknown Error/Invalid Error Code";
+}
```

Comment:
In a multiprocess environment, I've been using the remotery error code `RMT_ERROR_SOCKET_BIND_FAIL`

When creating my Remotery instances, I check whether this occurs and if it does not I bump the port number by one and try again.

This let me monitor the two processes that are spawned by the program.

.. Checking against an error string would be doable but cumbersome as I wouldn't be sure anymore what kind of error occurred.